modules/pipeline/services/crondsvc/crond.go

```diff
@@ -14,43 +14,98 @@
 package crondsvc
 
 import (
-	"context"
 	"fmt"
+	"strconv"
+	"strings"
 
 	"github.com/sirupsen/logrus"
 
 	"github.com/erda-project/erda/modules/pipeline/conf"
-	"github.com/erda-project/erda/modules/pipeline/spec"
 	"github.com/erda-project/erda/pkg/cron"
-	"github.com/erda-project/erda/pkg/jsonstore/storetypes"
+	"github.com/erda-project/erda/pkg/strutil"
 )
 
-const etcdCrondUpdateWatchKey = "/devops/pipeline/crond/update"
+const cronChanPrefixDeleteKey = "delete-"
+const cronChanPrefixAddKey = "add-"
 
+// Listen the key in the channel
+// decide to delete or add a scheduled task according to the prefix of the key
 func (s *CrondSvc) ListenCrond(pipelineCronFunc func(id uint64)) {
-	logrus.Info("crond: start listen")
 	for {
-		// watch: isPrefix[true], filterDelete[true], keyOnly[true]
-		err := s.js.IncludeWatch().Watch(context.Background(), etcdCrondUpdateWatchKey, false, true, true, nil,
```

Comment:
etcd is required for multi pipeline instance.
Imagine instance1 received add http request but nothing to do with instance2.

Reply:
Multi-instance watch etcd key to add or delete key